### DISCUSSION TITLE
Why rehype- and not remark- mathjax

### DISCUSSION THREAD

**Discussion by ChauhanT** [remarkjs/remark#777] [Q&A]
Hi, I'm curious as to why remark-math uses rehype mathjax instead of going directly from remark to chtml. I want to try and implement states which will allow me to use \ref blocks within a page (gatsbyjs), but can't figure out how. I had done this previously for mathjax-node. But mathjax-node uses mathjax 2, while remark-mathjax uses mathjax 3 and I would like to go with it instead.

Thanks!

**Comment by ChristianMurphy** [comment]
Some context, remark plugins work specifically on markdown, not HTML. rehype works specifically on HTML, not markdown.
`remark-math` recognizes blocks of math in the document (https://astexplorer.net/#/gist/4deae2c025b610234be53373f78eb28c/1495579f62858e1b07df0690346dca70d10685b3), it can then be handed off to `rehype-mathjax` which renders the math and can use chtml.

> I want to try and implement states which will allow me to use \ref blocks within a page (gatsbyjs), but can't figure out how.

Have you checked if MathJax 3 supports the syntax you are using?
For example trying it on the MathJax demo https://www.mathjax.org/#demo

Also have you checked that you've passed rehype-mathjax the same settings you passed to mathjax-node?
https://github.com/remarkjs/remark-math/tree/main/packages/rehype-mathjax#api

**Reply by smalltimer** [reply]
First, thank you for the blazing quick response. Really appreciated. 

*Point 1*: Yes, Mathjax 3 supports the syntax. Thank you for that link, very useful for the future. I have tested using: 
```
Here is a display equation with a tag and a label: 
$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}. \tag{1} \label{eq1}$$

This can be referred to using $\eqref{eq1}$. It works!
```

*Point 2*: I have passed tex options using `options.tex`, and they work. In the example in Point 1 above, this would translate to an anchor link for equation 1. 

*The problem*: The snippet in Point 1 allows me to nicely illustrate the problem. Since each math content in the snippet (this would be one display-math and one inline-math) is typset separately, the nodes do not remember each others' state. Thus, the `\eqref` tag is not aware of the display-math equation with `\label{eq1}`. So it just ends up throwing a `???` inside the brackets instead. I hope I was able to explain this properly. If not, please let me know and I can also share some code. Thank you very much.
